x/incentive/keeper/genesis.go

```diff
@@ -49,6 +51,80 @@ func (k Keeper) InitGenesis(ctx context.Context, gs types.GenesisState) error {
 		}
 	}
 
+	for _, entry := range gs.RefundableMsgHashes {
+		// hashes are hex encoded for better readability
+		bz, err := hex.DecodeString(entry)
+		if err != nil {
+			return fmt.Errorf("error decoding msg hash: %w", err)
+		}
+		if err := k.RefundableMsgKeySet.Set(ctx, bz); err != nil {
+			return fmt.Errorf("error storing msg hash: %w", err)
+		}
+	}
+
+	for _, entry := range gs.FinalityProvidersCurrentRewards {
+		// check that fp address exists
+		fpAddr := sdk.MustAccAddressFromBech32(entry.Address)
+		acc := k.accountKeeper.GetAccount(ctx, fpAddr)
+		if acc == nil {
+			return fmt.Errorf("finality provider account with address %s does not exist", entry.Address)
+		}
+		if err := k.FinalityProviderCurrentRewards.Set(ctx, fpAddr.Bytes(), *entry.Rewards); err != nil {
+			return err
+		}
+	}
+
+	for _, entry := range gs.FinalityProvidersHistoricalRewards {
+		// check that fp address exists
+		fpAddr := sdk.MustAccAddressFromBech32(entry.Address)
+		acc := k.accountKeeper.GetAccount(ctx, fpAddr)
+		if acc == nil {
+			return fmt.Errorf("finality provider account with address %s does not exist", entry.Address)
+		}
+		if err := k.FinalityProviderHistoricalRewards.Set(
+			ctx,
+			collections.Join(fpAddr.Bytes(), entry.Period),
+			*entry.Rewards,
+		); err != nil {
+			return err
+		}
+	}
+
+	for _, entry := range gs.BtcDelegationRewardsTrackers {
+		// check that fp and delegator accounts exists
+		fpAddr := sdk.MustAccAddressFromBech32(entry.FinalityProviderAddress)
+		acc := k.accountKeeper.GetAccount(ctx, fpAddr)
+		if acc == nil {
+			return fmt.Errorf("finality provider account with address %s does not exist", entry.FinalityProviderAddress)
+		}
+		delAddr := sdk.MustAccAddressFromBech32(entry.DelegatorAddress)
+		acc = k.accountKeeper.GetAccount(ctx, delAddr)
+		if acc == nil {
+			return fmt.Errorf("delegator account with address %s does not exist", entry.DelegatorAddress)
+		}
+		if err := k.BTCDelegationRewardsTracker.Set(
```

Comment:
can we use the function `setBTCDelegationRewardsTracker`

Reply:
We can use that function. Initially I chose this approach because it separates the storing of the `BTCDelegationRewardsTracker` and the `BTCDelegatorToFP` entries. But considering we're checking the addresses maps in the `Validate()` we can use that function

```
// setBTCDelegationRewardsTracker sets a new structure in the store, it fails and returns an error if the rwd fails to marshal.
func (k Keeper) setBTCDelegationRewardsTracker(ctx context.Context, fp, del sdk.AccAddress, rwd types.BTCDelegationRewardsTracker) error {
	k.setBTCDelegatorToFP(ctx, del, fp)
	return k.BTCDelegationRewardsTracker.Set(ctx, collections.Join(fp.Bytes(), del.Bytes()), rwd)
}
```